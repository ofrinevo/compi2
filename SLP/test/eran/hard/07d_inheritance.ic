class A { 
	int x ; 
	int y ; 
	
	void setX (int x1) { 
		x = x1 ; 
	}
	void setY (int y1) { 
		this.y = y1 ; 
	}
	void print () { 
		Library.print("A: [") ; 
		Library.printi(x) ; 		
		Library.print(", ") ; 
		Library.printi(this.y) ; 
		Library.print("] ") ;
		Library.println("") ;  
	}
}

class B extends A { 
	void print() { 
		Library.println("**************") ; 
		Library.print("* B: [") ; 
		Library.printi(x) ; 		
		Library.print(", ") ; 
		Library.printi(this.y) ; 
		Library.print("] *") ;
		Library.println("") ;  
		Library.println("**************") ; 
	}
}

class C extends B { 
	int z ; 
	
	void setZ (int z) { 
		this.z = z ; 
	}

	void print() { 
		Library.println("******************") ; 
		Library.print("* C: [") ; 
		Library.printi(x) ; 		
		Library.print(", ") ; 
		Library.printi(this.y) ; 
		Library.print(", ") ; 
		Library.printi(z) ; 
		Library.print("] *") ;
		Library.println("") ;  
		Library.println("******************") ; 
	}
}

class MainClass { 

	static void main (string[] args) { 
		A a = new A() ; a.setX(4); a.setY(8); 	
		A b = new B() ; b.setX(-2); b.setY(5); 
		A c = new C() ; c.setX(-3); c.setY(-3); 
		
		a.print() ; 
		b.print() ; 
		c.print() ; 
	}

}