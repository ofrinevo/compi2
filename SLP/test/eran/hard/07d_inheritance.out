A: [4, 8] 
**************
* B: [-2, 5] *
**************
******************
* C: [-3, -3, 0] *
******************
